src/jit/morph.cpp

```diff
@@ -3185,7 +3185,16 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
         }
 
 #ifdef _TARGET_ARM_
-
+#ifndef LEGACY_BACKEND
+        if (reMorphing)
+        {
+            passUsingFloatRegs = isValidFloatArgReg(argEntry->regNum);
+        }
+        else
+        {
+            passUsingFloatRegs = !callIsVararg && (isHfaArg || varTypeIsFloating(argx));
+        }
+#else  // LEGACY_BACKEND
         bool passUsingIntRegs;
```

Comment:
Why don't you need to execute the following code (now under `LEGACY_BACKEND`) for RyuJIT/arm32? E.g., don't you need to check `!opts.compUseSoftFP`? And what about the rules implemented below about argument register alignment for long and double? It seems like you are altering the calling ABI.